core/src/test/java/org/infinispan/functional/FunctionalEncoderTest.java

```diff
@@ -0,0 +1,29 @@
+package org.infinispan.functional;
+
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.cache.StorageType;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "functional.FunctionalEncoderTest")
+public class FunctionalEncoderTest extends FunctionalMapTest {
```

Comment:
I think that we need a test using the `AdvancedCache.withEncoders`, too. I am not convinced that all the encodings/decodings happen at the right places. You should also test transactional case (as you've missed encoding on Mutations), so basing this just on FunctionalMapTest is not sufficient.
E.g. I wonder why the testsuite did not catch the `Ids.BI_FUNCTION_MAPPER`, could you check?